### PR TITLE
Input text branch added partially complete

### PR DESCRIPTION
**Pull request by mukuagar**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

## Sample Card
If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/macos/ADCMacOSVisualizer/ADCMacOSVisualizer/MainViewController.swift [hunk 1]
```diff
@@ -18,7 +18,7 @@ class MainViewController: NSViewController {
     
     // MARK: Actions
     @objc private func handleButtonAction() {
-        let result = AdaptiveCard.from(jsonString: testPayload)
+        let result = AdaptiveCard.from(jsonString: simpleTextPayload)// testPayload)
         guard let card = result.getAdaptiveCard() else {
             print("RENDER FAILED")
```
Comment (blue-stone99) [inline source/macos/ADCMacOSVisualizer/ADCMacOSVisualizer/MainViewController.swift:23]: Can we revert these changes back?

File: source/macos/ADCMacOSVisualizer/ADCMacOSVisualizer/MainViewController.swift [hunk 2]
```diff
@@ -89,6 +89,18 @@ let simpleTextPayload = """
...
+        "placeholder": "Placeholder text",
+        "id": "lol"
+    },
+    {
+        "type": "Input.Text",
+        "placeholder": "Placeholder text",
+        "id": "lol_again",
+        "value": "Mukul Agarwal",
+        "isMultiline": true
     }
```
Comment (blue-stone99) [inline source/macos/ADCMacOSVisualizer/ADCMacOSVisualizer/MainViewController.swift:104]: This one as well?

File: source/macos/AdaptiveCards/AdaptiveCards-bridge/AutoGenerated/Models/ACSTextInput.mm [hunk 3]
```diff
@@ -140,8 +140,60 @@ - (void)setRegex:(NSString * _Nonnull)value
...
+//  return self;
+//}
+//
+//- (void)setMaximumLength:(int)len {
+//  maxLength = len;
+//}
+//
+//- (int)maximumLength {
+//  return maxLength;
+//}
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards-bridge/AutoGenerated/Models/ACSTextInput.mm:166]: All these comments can be removed.
Ideally we shouldn't change anything in ACS* files until or unless really required.

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift [hunk 4]
```diff
@@ -0,0 +1,81 @@
+//
+//  TextInputRenderer.swift
+//  AdaptiveCards
+//
+//  Created by mukuagar on 11/02/21.
+//
+
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift:7]: This header can be removed as well.

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        }
+        var doesPlaceholderExist = false
+        var doesValueExist = false
+        // Check if placeholder and initial value exist
+        if inputBlock.getPlaceholder() != nil {
+            doesPlaceholderExist = true
+            }
+        if inputBlock.getValue() != nil {
+            doesValueExist = true
+            }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift:51]: These can written like,
```suggestion
        let doesPlaceholderExist = inputBlock.getPlaceholder() != nil
        let doesValueExist = inputBlock.getValue() != nil
```

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                attributedPlaceHolder.addAttributes([.foregroundColor: textColor], range: NSRange(location: 0, length: attributedPlaceHolder.length))
+            }
+            textView.placeholderAttributedString = attributedPlaceHolder
+        }
+        if doesValueExist {
+            attributedInitialValue = NSMutableAttributedString(string: inputBlock.getValue() ?? "")
+            
+            textView.attributedStringValue = attributedInitialValue
+            }
+        return textView
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift:65]: Lint issue: Spacing and horizontal line.

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    var maxLen: Int = 0
+    override func textDidChange(_ notification: Notification) {
+        if maxLen > 0 {
+            guard let object = notification.object as? NSTextView else {
+                return
+            }
+                if (object.string.count) > maxLen {
+                    object.string = String(object.string.dropLast())
+            }
+        }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift:79]: Can we wrap it inside a guard statements instead?
```suggestion
    override func textDidChange(_ notification: Notification) {
        guard maxLen > 0  else { return } // Add approprite comment
        guard let textView = notification.object as? NSTextView, textView.string.count > maxLen else { return }
        textView.string = String(textView.string.dropLast())
```

**Comment by mukuagar** [comment]
Closing PR
